build_tools/cmake/iree_bytecode_module.cmake

```
@@ -108,13 +108,23 @@ function(iree_bytecode_module)
   list(APPEND _ARGS "-o")
   list(APPEND _ARGS "${_MODULE_FILE_NAME}")
 
-  # If an LLVM CPU backend is enabled, supply the linker tool.
+  # If an LLVM CPU backend is enabled, supply the embedded linker tool.
+  # Note: "lld" requires a -flavor flag. This is passed in
+  # EmbeddedLinkerTool.cpp.
   if(IREE_LLD_TARGET)
     iree_get_executable_path(_LINKER_TOOL_EXECUTABLE "lld")
-    list(APPEND _ARGS "-iree-llvm-system-linker-path=\"${_LINKER_TOOL_EXECUTABLE}\"")
     list(APPEND _ARGS "-iree-llvm-embedded-linker-path=\"${_LINKER_TOOL_EXECUTABLE}\"")
   endif()
 
+  # Support testing in TSan build dirs. Unlike other sanitizers, TSan is an
+  # ABI break: when the host code is built with TSan, the module must be too,
+  # otherwise we get crashes calling module code.
+  if (IREE_ENABLE_TSAN_IN_TEST_MODULES)
+    list(APPEND _ARGS "-iree-llvm-sanitize=thread")
+    # Avoid undefined-symbol errors on tsan symbols e.g. __tsan_read8
+    list(APPEND _ARGS "-iree-llvm-link-embedded=false")
```

Let's add a bit more to this comment, since this is a behavior change.

```suggestion
    # TSan requires some imported symbols like __tsan_read8, so
    # we must use the system linker with shared (not static) linking.
    # This changes behavior for tests that were expecting to use
    # the embedded linker, but at least we get some coverage
    # using TSan this way.
    list(APPEND _ARGS "-iree-llvm-link-embedded=false")
```